pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/columnminmaxvalue/ColumnMinMaxValueGenerator.java

```diff
@@ -124,227 +131,251 @@ private List<String> getColumnsToAddMinMaxValue() {
+              float[] values = rawIndexReader.getFloatMV(docId, readerContext);
+              for (float value : values) {
                 min = Math.min(min, value);
                 max = Math.max(max, value);
               }
-            } else {
-              for (int docId = 0; docId < numDocs; docId++) {
-                float[] values = rawIndexReader.getFloatMV(docId, readerContext);
-                for (float value : values) {
-                  min = Math.min(min, value);
-                  max = Math.max(max, value);
-                }
-              }
             }
-            SegmentColumnarIndexCreator.addColumnMinMaxValueInfo(_segmentProperties, columnName, String.valueOf(min),
-                String.valueOf(max), storedType);
-            break;
           }
-          case DOUBLE: {
-            double min = Double.POSITIVE_INFINITY;
-            double max = Double.NEGATIVE_INFINITY;
-            if (isSingleValue) {
-              for (int docId = 0; docId < numDocs; docId++) {
-                double value = rawIndexReader.getDouble(docId, readerContext);
+          minValue = min;
+          maxValue = max;
+          break;
+        }
+        case DOUBLE: {
+          double min = Double.POSITIVE_INFINITY;
+          double max = Double.NEGATIVE_INFINITY;
+          if (isSingleValue) {
+            for (int docId = 0; docId < numDocs; docId++) {
+              double value = rawIndexReader.getDouble(docId, readerContext);
+              min = Math.min(min, value);
+              max = Math.max(max, value);
+            }
+          } else {
+            for (int docId = 0; docId < numDocs; docId++) {
+              double[] values = rawIndexReader.getDoubleMV(docId, readerContext);
+              for (double value : values) {
                 min = Math.min(min, value);
                 max = Math.max(max, value);
               }
-            } else {
-              for (int docId = 0; docId < numDocs; docId++) {
-                double[] values = rawIndexReader.getDoubleMV(docId, readerContext);
-                for (double value : values) {
-                  min = Math.min(min, value);
-                  max = Math.max(max, value);
-                }
-              }
             }
-            SegmentColumnarIndexCreator.addColumnMinMaxValueInfo(_segmentProperties, columnName, String.valueOf(min),
-                String.valueOf(max), storedType);
-            break;
           }
-          case STRING: {
-            String min = null;
-            String max = null;
-            if (isSingleValue) {
-              for (int docId = 0; docId < numDocs; docId++) {
-                String value = rawIndexReader.getString(docId, readerContext);
+          minValue = min;
+          maxValue = max;
+          break;
+        }
+        case BIG_DECIMAL: {
+          Preconditions.checkState(isSingleValue, "Unsupported multi-value BIG_DECIMAL column: %s", columnName);
```

Comment:
is this generally true in Pinot or just too complicated to handle here for now?

Reply:
MV is only allowed within `DIMENSION`, and it is very rare to configure `BIG_DECIMAL` as dimension. In the future we can consider adding MV support, but as of now we don't have APIs to read MV BigDecimal values